cumulusci/cli/cci.py

```diff
@@ -1345,30 +1343,56 @@ def task_list(runtime, plain, print_json):
 
 
 @task.command(name="doc", help="Exports RST format documentation for all tasks")
-@pass_runtime(require_project=False)
-def task_doc(runtime):
-    config_src = runtime.universal_config
-
-    click.echo("==========================================")
-    click.echo("Tasks Reference")
-    click.echo("==========================================")
-    click.echo("")
-
-    for name, options in config_src.tasks.items():
+@click.option("--project", "project", is_flag=True, help="Write project specific tasks")
+@click.option(
+    "--write",
+    "write",
+    is_flag=True,
+    help="If project flag is toggled will write to ./docs/project_tasks.rst otherwise will write by default to ./docs/cumulusci_tasks.rst",
+)
+@pass_runtime(
+    require_project=False,
+)
+def task_doc(runtime, project=False, write=False):
+    config = runtime.universal_config
+    tasks = config.tasks.items()
+    file_name = "./docs/cumulusci_tasks.rst"
+    # setting to read for project specific configuration
+    if project:
+        config = runtime.project_config.config
+        project_name = runtime.project_config.project__name.capitalize()
+        tasks = config["tasks"].items()
```

Comment:
I tried it and it's still including all tasks when I use the `--project` option. project_config.config contains the _merged_ config combined from all sources. I think you want project_config.config_project here.